src/core/src/core_logic/PatchInstaller.py

```diff
@@ -42,6 +43,10 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.last_still_needed_package_versions = None
         self.progress_template = "[Time available: {0} | A: {1}, S: {2}, F: {3} | D: {4}]\t {5}"
 
+        self.attempted_parent_update_count = 0
+        self.successful_parent_update_count = 0
+        self.failed_parent_update_count = 0
```

Comment:
Variable naming could be improved. <>_parent_update_count raises the question of: parent in terms what?

Reply:
These were the existing variable names in the code. These variables were local in the function install_updates. I just made them global because these variables are being referred in multiple functions now. Being said that I agree that the variable names can be improved.

Reply:
Should I leave as the variable names are already existing or is there any suggestion on name?

Reply:
As you mentioned, these were local to install_updates() function earlier, hence the naming was relatable. For the overall class, with a mixture of packages and updates being used intermittently, this could cause confusion. 

Also, to answer your other query on the file containing "patch", "package" and "update" terminologies. "patch" is generally referred in the sense of the overall operation, hence the term overall_patch_installation_successful. Even patch_installation_successful is used to compute the overall operation status. Reference to a comment on one of your previous PRs from Koshy: https://github.com/Azure/LinuxPatchExtension/pull/161#discussion_r1061048791

"package" should denote the package metadata and "update" any updates available for a package.

Following this, my suggestion would be to rename this to attempted_parent_package_update_count and similar for others.

Please also follow the same distinction between patch, package and updates elsewhere

Reply:
Another suggestion based on the usage below: attempted_parent_update_count can be attempted_parent_package_install_count

Reply:
Updated to attempted_parent_package_install_count and similar for other variables.
Sure, will follow the distinction as you explained going forward. Currently there is no other variable which needs to be renamed as per this distinction.